frontend/tests/controllers-test/challengeListCtrl.test.js

```diff
@@ -0,0 +1,285 @@
+            };
+            vm = createController();
+            expect(vm.pastList).toEqual(successResponse.results);
+            expect(vm.nonePastChallenge).toBeTruthy();
+        });
+
+        it('check description length and calculate timezone of past challenge `challenges/challenge/past`', function () {
+            isPastChallengeSuccess = true;
+            isPresentChallengeSuccess = true;
+            isUpcomingChallengeSucess = true;
+            successResponse = {
+                results: [
+                    {
+                        id: 1,
+                        description: "the length of the past challenge description is greater than or equal to 50",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Fri June 12 2018 22:41:51 GMT+0530",
+                        end_date: "Fri June 12 2099 22:41:51 GMT+0530"
+                    },
+                    {
+                        id: 2,
+                        description: "random description",
+                        creator: {
+                            id: 1
+                        },
+                        start_date: "Sat May 26 2015 22:41:51 GMT+0530",
+                        end_date: "Sat May 26 2099 22:41:51 GMT+0530"
+                    }
+                ]
+            };
+            vm = createController();
+            expect(vm.pastList).toEqual(successResponse.results);
+            expect(vm.nonePastChallenge).toBeFalsy();
+
+            var timezone = moment.tz.guess();
+            var zone = moment.tz.zone(timezone);
+            for (var i in vm.pastList) {
+                if (vm.pastList[i].description.length >= 50) {
+                    expect(vm.pastList[i].isLarge).toEqual("...");
+                } else {
+                    expect(vm.pastList[i].isLarge).toEqual("");
+                }
+                var offset = new Date(vm.pastList[i].start_date).getTimezoneOffset();
+                expect(vm.pastList[i].start_zone).toEqual(zone.abbr(offset));
+                offset = new Date(vm.pastList[i].end_date).getTimezoneOffset();
+                expect(vm.pastList[i].end_zone).toEqual(zone.abbr(offset));
+
+                expect(vm.challengeCreator[vm.pastList[i].id]).toEqual(vm.pastList[i].creator.id);
+                expect(utilities.storeData).toHaveBeenCalledWith("challengeCreator", vm.challengeCreator);
+            }
+            expect(utilities.hideLoader).toHaveBeenCalled();
+        });
+
+        it('past challenge backend error `challenges/challenge/past`', function () {
+            isPastChallengeSuccess = false;
+            isPresentChallengeSuccess = true;
+            isUpcomingChallengeSucess = true;
+            // success response for the ongoing and upcoming challenge
+            successResponse = {
+                results: []
+            };
+            errorResponse = {
+                error: 'error'
+            };

```

Comment:
You might want to validate these too

Reply:
as there is no code for this(^) in  `frontend/tests/controllers-test/challengeListCtrl.js`, therefore removed it.